### PR TITLE
test: refactor DeepSeek v3 MTP tests into standard two-step structure

### PR DESCRIPTION
**Pull request by parambole**
# Description

This PR refactors the DeepSeek v3 MTP end-to-end test into our standard two-step script structure, separating the CPU-intensive checkpoint conversion process from the TPU-based fine-tuning loop. This modularity improves test reliability and mirrors our standard testing environments where conversion and training run on different hardware profiles.

FIXES: b/502684535

# Tests

Verified Step 1 (Checkpoint Conversion) executes correctly on CPU.
Verified Step 2 (Fine-tuning) successfully completes the 10-step training loop on v5p-128 without OOM or parallelism errors.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    tokenizer_type=huggingface \
+    tokenizer_path=${TOKENIZER_PATH} \
+    attention=flash \
+    dtype=bfloat16 \
+    weight_dtype=bfloat16 \
+    opt_type=sgd \
+    enable_checkpointing=True \
+    mtp_num_layers=1 \
+    mtp_loss_scaling_factor=0.1 \
+    checkpoint_storage_concurrent_gb=1024 \
```
Comment (shuningjin) [inline tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh]: nit: remove trailing \

File: tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  # this bucket will store all the files generated by MaxText during a run
+  export BASE_OUTPUT_PATH=gs://runner-maxtext-logs/$(date +%Y-%m-%d-%H-%M)
+  echo "BASE_OUTPUT_PATH is not set"
+fi
+BASE_OUTPUT_PATH=${BASE_OUTPUT_PATH%/}
+echo using BASE_OUTPUT_PATH = ${BASE_OUTPUT_PATH}
+
+# Step 2:
+# We define the checkpoint paths. Pointing to the MTP bucket you populated earlier.
+SCANNED_CKPT_PATH=gs://maxtext-deepseek/deepseek3-671b-mtp/04-15-2026/scanned/0/items
```
Comment (shuningjin) [inline tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh]:
```
# Use a hard-coded golden checkpoint, rather than checkpoints generated by Step 1 as it is not in daily test.
```

File: tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+set -ex
+
+export MODEL_NAME='deepseek3-671b'
+export TOKENIZER_PATH='deepseek-ai/DeepSeek-V3'
+
+# Installing torch for checkpoint conversion and forward_pass_logit_checker.py
+python3 -m pip install torch --index-url https://download.pytorch.org/whl/cpu
+
+# e.g., $HOME/maxtext/src/maxtext
+export MAXTEXT_PKG_DIR="${MAXTEXT_PKG_DIR:-${MAXTEXT_REPO_ROOT:-$PWD}/src/maxtext}"
```
Comment (shuningjin) [inline tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh]: nit: can remove here. 

can you also remove that in the other file:
https://github.com/AI-Hypercomputer/maxtext/blob/9c68b1abd479c4e95fc6e66f0c7a4235a9507a9e/tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek.sh#L21-L22

File: tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+echo "************************"
+
+
+# 1. Run short fine-tuning to verify MTP training loop (10 steps)
+python3 -m maxtext.trainers.pre_train.train "${MAXTEXT_CONFIGS_DIR:-${MAXTEXT_REPO_ROOT:-$PWD}/src/maxtext/configs}"/base.yml \
+    base_output_directory=${BASE_OUTPUT_PATH} \
+    dataset_path=${DATASET_PATH} \
+    load_parameters_path=${SCANNED_CKPT_PATH} \
+    run_name=mtp_fine_tuning \
+    per_device_batch_size=4 \
```
Comment (shuningjin) [inline tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh]: train compile gives error. use `per_device_batch_size=1` is fine

**Review by shuningjin — COMMENTED** [review]
Thanks! Some minor comments.

**Review by shuningjin — APPROVED** [review]

File: tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh (hunk 4 above)
Comment (parambole) [reply tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh]: Great catch! I had updated in my local runs but forgot to do it here.

File: tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh (hunk 3 above)
Comment (parambole) [reply tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh]: Done!

File: tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh (hunk 1 above)
Comment (parambole) [reply tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh]: Done

File: tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+BASE_OUTPUT_PATH=${BASE_OUTPUT_PATH%/}
+echo using BASE_OUTPUT_PATH = ${BASE_OUTPUT_PATH}
+
+# Step 2:
+# Use a hard-coded golden checkpoint, rather than checkpoints generated by Step 1 as it is not in daily test.
+SCANNED_CKPT_PATH=gs://maxtext-deepseek/deepseek3-671b-mtp/04-15-2026/scanned/0/items
+# Non-Googlers please remember to point `DATASET_PATH` to the GCS bucket where you have your training data
+export DATASET_PATH=gs://maxtext-dataset
+
+echo "************************"
```
Comment (bvandermoon) [inline tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh:33]: Nit: Shouldn't this be removed? Was it just for testing?

File: tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+# 1. Run short fine-tuning to verify MTP training loop (10 steps)
```
Comment (bvandermoon) [inline tests/end_to_end/tpu/deepseek/v3-671b/2_test_deepseek_mtp.sh:35]: This should be 2, right?

File: tests/end_to_end/tpu/deepseek/v3-671b/1_test_deepseek_mtp.sh [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+# Assume HF checkpoints are uploaded to GCS bucket at CKPT_BUCKET 
+# Non-Googlers please remember to point `CKPT_BUCKET` to GCS buckets that you own
+# Copying the HF checkpoint into a local directory `/tmp` -- you are free to use a different directory
+if [ -z "${CKPT_DISK_LOCATION}" ]; then
+  export CKPT_BUCKET=gs://maxtext-deepseek/deepseek3-671b/hf
+  gcloud storage cp -r ${CKPT_BUCKET} /tmp
+  export CKPT_DISK_LOCATION=/tmp/hf
+fi
+
+# 1. Convert checkpoint to `scanned` format, more suitable for training
```
Comment (bvandermoon) [inline tests/end_to_end/tpu/deepseek/v3-671b/1_test_deepseek_mtp.sh:38]: Why `Step 1` on line 27, and then `1` again on line 38?

**Review by bvandermoon — APPROVED** [review]
